File: scratch/parse_article.js
const fs = require('fs');

const raw = fs.readFileSync('C:/Users/Josué/.gemini/antigravity-ide/brain/b602958c-4313-405a-83e7-7d7d9ec57c50/.system_generated/steps/39/content.md', 'utf-8');

// Strip script and style tags
let cleaned = raw.replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '');
cleaned = cleaned.replace(/<style\b[^<]*(?:(?!<\/style>)<[^<]*)*<\/style>/gi, '');

// Extract images
const imgRegex = /<img[^>]+src="([^">]+)"(?:[^>]+alt="([^">]*)")?/gi;
let match;
const images = [];
while ((match = imgRegex.exec(cleaned)) !== null) {
  images.push({ src: match[1], alt: match[2] || '' });
}

// Convert tags like <p>, <h1>-<h6>, <li> to line breaks
let text = cleaned
  .replace(/<(?:h[1-6]|p|li|tr|div)[^>]*>/gi, '\n')
  .replace(/<[^>]+>/g, '')
  .replace(/&nbsp;/g, ' ')
  .replace(/&amp;/g, '&')
  .replace(/&lt;/g, '<')
  .replace(/&gt;/g, '>')
  .replace(/\n\s*\n/g, '\n')
  .trim();

fs.writeFileSync('scratch/article_clean.txt', text, 'utf-8');
fs.writeFileSync('scratch/article_images.json', JSON.stringify(images, null, 2), 'utf-8');
console.log('Done! Text length:', text.length, 'Images found:', images.length);

File: scratch/parse_article.py
import re
from bs4 import BeautifulSoup

with open(r'C:\Users\Josué\.gemini\antigravity-ide\brain\b602958c-4313-405a-83e7-7d7d9ec57c50\.system_generated\steps\39\content.md', 'r', encoding='utf-8', errors='ignore') as f:
    text = f.read()

soup = BeautifulSoup(text, 'html.parser')
content_div = soup.find('div', class_=re.compile(r'betterdocs-entry-content|entry-content|post-content'))
if not content_div:
    content_div = soup.find('article')

with open('scratch/article_text.txt', 'w', encoding='utf-8') as out:
    target = content_div if content_div else soup
    for el in target.find_all(['h1', 'h2', 'h3', 'h4', 'p', 'li', 'table', 'img']):
        if el.name == 'img':
            src = el.get('src') or el.get('data-src') or ''
            alt = el.get('alt') or ''
            out.write(f"\n[IMG]: {src} | Alt: {alt}\n")
        elif el.name == 'table':
            out.write(f"\n[TABLE]:\n{el.get_text(' | ', strip=True)}\n")
        else:
            txt = el.get_text(' ', strip=True)
            if txt:
                out.write(f"{el.name.upper()}: {txt}\n")

print("Finished parsing article.")
